Misleading error message when connecting to an MQTT 3 only broker with MQTT 5

## Expected behavior
Handle error with a better exception 
## Actual behavior

```
Anjas-MacBook-Pro:hivemq-cli ahelmbre$ java -jar build/libs/mqtt-cli-1.0.0.jar pub -pw test -u anja -t test -m ";2Cmessage" -i test -d 
Client test: sending CONNECT
Client test: com.hivemq.client.mqtt.mqtt5.exceptions.Mqtt5DisconnectException: Exception while decoding CONNACK: remaining length too short
	at com.hivemq.client.internal.mqtt.MqttBlockingClient.connect(MqttBlockingClient.java:91)
	at com.hivemq.cli.mqtt.MqttClientExecutor.mqtt5Connect(MqttClientExecutor.java:79)
	at com.hivemq.cli.mqtt.AbstractMqttClientExecutor.connectMqtt5Client(AbstractMqttClientExecutor.java:238)
	at com.hivemq.cli.mqtt.AbstractMqttClientExecutor.connect(AbstractMqttClientExecutor.java:201)
	at com.hivemq.cli.mqtt.MqttClientExecutor.connect(MqttClientExecutor.java:62)
	at com.hivemq.cli.mqtt.AbstractMqttClientExecutor.getMqttClientFromCacheOrConnect(AbstractMqttClientExecutor.java:432)
	at com.hivemq.cli.mqtt.AbstractMqttClientExecutor.publish(AbstractMqttClientExecutor.java:110)
	at com.hivemq.cli.mqtt.MqttClientExecutor.publish(MqttClientExecutor.java:62)
	at com.hivemq.cli.commands.cli.PublishCommand.run(PublishCommand.java:109)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1729)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2101)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2068)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1935)
	at picocli.CommandLine.execute(CommandLine.java:1864)
	at com.hivemq.cli.MqttCLIMain.main(MqttCLIMain.java:72)
Caused by: com.hivemq.client.mqtt.exceptions.MqttDecodeException: Exception while decoding CONNACK: remaining length too short
Client test: Exception while decoding CONNACK: remaining length too short

```


## To Reproduce

* try to publish  with an Mqtt 5 Client 
* broker is set to a mosquitto broker

### Steps

### Reproducer code

## Details
- Affected HiveMQ MQTT Client version(s): 1.1.2
- Used JVM version:1.8

--- a/src/main/java/com/hivemq/client/internal/mqtt/codec/decoder/mqtt5/Mqtt5ConnAckDecoder.java
+++ b/src/main/java/com/hivemq/client/internal/mqtt/codec/decoder/mqtt5/Mqtt5ConnAckDecoder.java
@@ -26,6 +26,7 @@ import com.hivemq.client.internal.mqtt.message.connect.connack.MqttConnAck;
 import com.hivemq.client.internal.mqtt.message.connect.connack.MqttConnAckRestrictions;
 import com.hivemq.client.internal.util.collections.ImmutableList;
 import com.hivemq.client.mqtt.datatypes.MqttQos;
+import com.hivemq.client.mqtt.mqtt3.message.connect.connack.Mqtt3ConnAckReturnCode;
 import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAckReasonCode;
 import com.hivemq.client.mqtt.mqtt5.message.disconnect.Mqtt5DisconnectReasonCode;
 import io.netty.buffer.ByteBuf;
@@ -62,7 +63,7 @@ public class Mqtt5ConnAckDecoder implements MqttMessageDecoder {
         checkFixedHeaderFlags(FLAGS, flags);
 
         if (in.readableBytes() < MIN_REMAINING_LENGTH) {
-            throw remainingLengthTooShort();
+            return tryDecodeMqtt3(in);
         }
 
         final short connAckFlags = in.readUnsignedByte();
@@ -260,5 +261,18 @@ public class Mqtt5ConnAckDecoder implements MqttMessageDecoder {
         return new MqttConnAck(reasonCode, sessionPresent, sessionExpiryInterval, serverKeepAlive,
                 assignedClientIdentifier, enhancedAuth, restrictions, responseInformation, serverReference,
                 reasonString, userProperties);
+    }
+
+    public @NotNull MqttConnAck tryDecodeMqtt3(final @NotNull ByteBuf in) throws MqttDecoderException {
+        if (in.readableBytes() == 2) {
+            in.readUnsignedByte(); // ignore connAckFlags
+            final Mqtt3ConnAckReturnCode returnCode = Mqtt3ConnAckReturnCode.fromCode(in.readUnsignedByte());
+            if (returnCode == Mqtt3ConnAckReturnCode.UNSUPPORTED_PROTOCOL_VERSION) {
+                return new MqttConnAck(Mqtt5ConnAckReasonCode.UNSUPPORTED_PROTOCOL_VERSION, false,
+                        SESSION_EXPIRY_INTERVAL_FROM_CONNECT, KEEP_ALIVE_FROM_CONNECT, null, null,
+                        MqttConnAckRestrictions.DEFAULT, null, null, null, MqttUserPropertiesImpl.NO_USER_PROPERTIES);
+            }
+        }
+        throw remainingLengthTooShort();
     }
 }

